### PR TITLE
experimental: add extended cmake options build heuristic

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Sometimes projects will have default options on that can cause build issues, and being able to switch these off enable builds to succeed. This cmake option adds the ability to run a CMake run listing all options available, and enables turning off those default values that are ON.

### CODE REVIEW

File: experimental/c-cpp/build_generator.py [hunk 1]
```diff
@@ -298,6 +299,120 @@ def name(self):
...
+    build_container = AutoBuildContainer()
+    build_container.list_of_commands = cmds_to_exec_from_root
+    build_container.heuristic_id = self.name + '1'
+    yield build_container
+
+  @property
+  def name(self):
+    return 'autogen-ConfMakeOpt'
+
+  def match_files(self, file_list: List[str]) -> None:
```
Comment (oliverchang) [inline experimental/c-cpp/build_generator.py:330]: Can we add a docstring here?

File: experimental/c-cpp/build_generator.py [hunk 2]
```diff
@@ -52,10 +53,10 @@ def match_files(self, file_list):
 
   def is_matched(self):
     """Returns True if the build heuristic found matching files."""
-    for matches in self.matches_found:
-      if len(matches) == 0:
-        return False
-    return True
+    for found_matches in self.matches_found.values():
+      if len(found_matches) > 1:
```
Comment (oliverchang) [inline experimental/c-cpp/build_generator.py:57]: Why > 1 and not > 0? Can you add a comment here?

**Review by oliverchang — APPROVED** [review]

File: experimental/c-cpp/build_generator.py (hunk 2 above)
Comment (DavidKorczynski) [reply experimental/c-cpp/build_generator.py]: It should be > 0 -- thanks for catching!

File: experimental/c-cpp/build_generator.py (hunk 1 above)
Comment (DavidKorczynski) [reply experimental/c-cpp/build_generator.py:330]: done

**Comment by DavidKorczynski** [comment]
No need for trial build in experimental

**Pull request merged**